src/drivers/vmount/output_rc.cpp

```diff
@@ -73,9 +73,13 @@ int OutputRC::update(const ControlData *control_data)
 
 	actuator_controls_s actuator_controls;
 	actuator_controls.timestamp = hrt_absolute_time();
-	actuator_controls.control[0] = _angle_outputs[0] / M_PI_F;
-	actuator_controls.control[1] = _angle_outputs[1] / M_PI_F;
-	actuator_controls.control[2] = _angle_outputs[2] / M_PI_F;
+	// _angle_outputs are in radians, actuator_controls are in [-1, 1]
+	actuator_controls.control[0] = (_angle_outputs[0] + _config.roll_offset * M_DEG_TO_RAD_F) /
+				       (_config.roll_range * (M_DEG_TO_RAD_F / 2.0f));
```

Comment:
why `/2.0f`?

Reply:
The purpose of the angle range parameters is to generalise this to support servos, rather than just outputting into brushless gimbal controllers. A standard servo has a movement range of somewhere between 60 and 180 degrees from MIN_PWM to MAX_PWM, so this control output calculation needs to take that into account. See the Ardupilot documentation of their version [here](http://ardupilot.org/copter/docs/common-camera-gimbal.html).

For the factor of 2 specifically: for a 90 degree servo this expression needs to be a linear transformation from [-45, 45] to [-1, 1] (ignoring the offset), hence dividing by range/2.

Reply:
I see, thanks! How problematic is it if you output something that's outside of that range? Is it worth guarding against ever having `actuator_controls.control[i]` > 1 or <-1?

Reply:
Won't the mixer saturate at 1 and -1?

What happens to a servo at and beyond the edge of its range seems to vary; of the two I tested, an expensive one saturated on its own and a cheap one jerked around.

It's certainly possible that someone could want to set upper and lower limits on a particular channel (e.g. to prevent a camera from pointing back at the frame, or if there are clearance constraints). I feel it's probably fine to just expect people to tweak their mixers for that though, since it's an airframe-dependent setting you only have to set once. The parameters I've added are the ones you need to iteratively tweak until the gimbal stabilizes correctly, so it would be quite inconvenient to do that with mixers. What do you think?